state/core_access.go

```diff
@@ -124,9 +119,15 @@ func (ca *CoreAccessor) Start(ctx context.Context) error {
 
 	// dial given celestia-core endpoint
 	endpoint := fmt.Sprintf("%s:%s", ca.coreIP, ca.grpcPort)
+	grpcOpts := make([]grpc.DialOption, 0)
+	if ca.enableTLS {
+		grpcOpts = append(grpcOpts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})))
```

Comment:
Without specifying valid certificates in the tls.Config object will not secure gRPC connection. By default, an empty tls.Config{} means that no certificate or private key is provided, which leads to an insecure setup.

Node already has TLS certificates handling logic [here](https://github.com/celestiaorg/celestia-node/blob/ace6840bb00b337c320c07be4dbe10e4d853271d/nodebuilder/p2p/tls.go#L24) for web socket, so you can reuse the certificate for the grpc connection.

Reply:
This was a proof of concept to get the Light node working against a Quicknode celestia endpoint. QN doesn't support client certs. gRPC auth for a QN endpoint occurs over the `x-token` header

Reply:
One more point here is the solution is not generic. You may not support the certificates but if someone does, the connection won't be established. Also, it is recommended to manually specify the MinTLSVersion(linter confirmed it)